sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -532,7 +545,7 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
 
   private var extraOptions = CaseInsensitiveMap[String](Map.empty)
 
-  private var foreachWriter: ForeachWriter[T] = null
+  private var foreachWriter: ForeachWriter[Any] = null
```

Comment:
@hvanhovell Hi Herman, can you signoff this change? It can be viewed together with this PR for more context: 
https://github.com/apache/spark/pull/41129/files#r1194152295

Thanks in advance!